### ISSUE TITLE
Web accessibility changes

### ISSUE THREAD

**Issue by smalers** [OpenCDSS/cdss-app-tstool-doc-user#121]
The State has indicated that documentation needs to be updated for ADA compliance.  The following are specific changes.

## Update CSS to underline links ##

This is needed for people that are colorblind because they may not be able to see that links are colored.  The necessary CSS for MkDocs custom CSS file is:

```
a:link {
  text-decoration: underline;
}
```

**Question**:  Does the underline need to be in black so that it is visible?  Underlining in the same color does not address the original visual issue.

## Add human-readable alt text for images ##

This is needed because readers that convert HTML to spoken language need to be able to describe the image,  This accommodates people with blindness.

I have been instructed that the alt text should be the same as the caption associated with the image.

**Question**:  Do the readers read the alt text and then also read the caption under the image, which is often just centered text?  If so, then a blind person would hear redundant words.

## Make sure there are no empty sections ##

This is perhaps not an ADA thing?

TSTool command documentation often has empty **Troubleshooting** sections and sometimes also **See Also**.  I'd like to keep the sections consistent so that users can expect that consistency.  My suggestion is just to link to the main TSTool troubleshooting page and I can probably always find something to put in **See Also**.

**Comment by smalers** [comment]
For underlines I guess it makes sense.  The color may still be hard for someone to fully see but if they can see the underline in some form they know it is a link.

For image alt text is it enough to say, for example, "Command editor" (so it will read as "command editor image").  Using the full name such as "ReadHydroBase command editor" is more descriptive but perhaps unnecessary given that the command name will have been read elsewhere and the scoring probably won't care.  I can add more detail if there are multiple images such as tabs on an editor.

As for the headers and closing tags, responding to specific reports on this will be necessary so I'll try to review the SiteImprove output.  There is a lot of TSTool documentation and such things may occur.

**Comment by Carolyn-CWCB** [comment]
Alt text needs to be unique on a page (relative to other alt text), so multiple images can't just say Command Editor. The scoring will put that up as a separate issue. In all of the StateDMI images on the commands pages, I typically added "command editor" to the alt text so that it did end up being like "ReadHydroBase command editor".

There are less than a dozen TSTool pages each with issues for headers and closing tags, so it shouldn't be as big a deal as the StateDMI command pages were where every page had a header issue.

**Comment by smalers** [comment]
I added the underline for links.

I added to all the troubleshooting sections.

I am now working through the alt text when I can stand it.  One issue came up.  On the release notes I use the same image many times, for example to show a bug icon.  What do I do in that case?  Here is an example.  It seems like the site checker needs ways to exclude certain pages or files because it is just noise.

```
* ![bug](bug.png) [11.13.00] Datastore configuration files for USGS NWIS and NRCS AWDB web services have been changed to use https URLs based on changes in the web services.
```

**Comment by smalers** [comment]
I finished updating the alt text for images for all content.  See the following:

[https://opencdss.state.co.us/tstool/14.7.0/doc-user/](https://opencdss.state.co.us/tstool/14.7.0/doc-user/)

As far as I know I am done with the updates other than the release notes icon issue above.  Once I know what to do with that the documentation will be done.  I need to run the automated tests on 14.7.0 and then I can release.

**Comment by Carolyn-CWCB** [comment]
> I added the underline for links.
> 
> I added to all the troubleshooting sections.
> 
> I am now working through the alt text when I can stand it. One issue came up. On the release notes I use the same image many times, for example to show a bug icon. What do I do in that case? Here is an example. It seems like the site checker needs ways to exclude certain pages or files because it is just noise.
> 
> ```
> * ![bug](bug.png) [11.13.00] Datastore configuration files for USGS NWIS and NRCS AWDB web services have been changed to use https URLs based on changes in the web services.
> ```

So it's the same image every time? Then don't worry about changing it.

**Comment by smalers** [comment]
OK, then the TSTool documentation is ready for deployment once I do testing on 14.7.0 and can make a release to the OpenCDSS website.

Yes, the release notes will use "bug", "change", "new", etc. many times on a single page.

Carolyn, can you paste the SiteImprove URL here for where I would look for those specific issues with TSTool sections and closed tags?  Or email me.  Then I can fix those.

Also, is there a way to embed something in the HTML to cause the site checker to ignore a page, or ignore specific checks on a page, for example the redundant alt text in release notes?  It does not seem like the type of check can be configured/excluded at a page level.  However, it does seem like entire pages and trees can be excluded from the crawl, such as old documentation releases [can be excluded](https://help.siteimprove.com/support/solutions/articles/80000863906) so that we can focus on recent and current documentation going forward.  Someone at the State would need to configure that.

Also, the web pages look busier with the underlines.  There is no way around that.  I'll get used to it over time.

**Comment by smalers** [comment]
Carolyn provided me with the SiteImprove link to fix empty elements and I did that.  If there are any other loose ends I can follow up.  I'll ask whether I should publish to "latest" so the software will review.

**Comment by smalers** [comment]
I'm closing this issue.  Documentation has been updated for accessibility.
